shared/styles/mixins.scss

```diff
@@ -14,82 +14,65 @@
+    font-size: #{$unitless-min}px;
+    font-size: calc((#{$font-multiplier} * #{100vw} + (#{$font-baseline}px)) * #{var(--scale-font)});
   }
 
   @media (min-width: $limit) {
-    font-size: #{$size}px;
-    font-size: calc(#{$size}px * #{var(--scale-multiplier)});
+    font-size: #{$unitless-size}px;
+    font-size: calc(#{$unitless-size}px * #{var(--scale-font)});
   }
 }
 
-@mixin viewport($media) {
-  @warn 'viewport mixin is deprecated and will be removed in a future version';
-
-  @if $media == handset {
-    @media only screen and (max-width: $min-600) { @content; }
-  }
-
-  @else if $media == tablet {
-    @media only screen and (min-width: $min-tablet) { @content; }
-  }
-
-  @else if $media == desktop {
-    @media only screen and (min-width: $min-desktop) { @content; }
-  }
-
-  /* stylelint-disable */
-  @else if $media == retina {
-    @media only screen and (-webkit-min-device-pixel-ratio: 1.3),
-      only screen and (-o-min-device-pixel-ratio: 13/10),
-      only screen and (min-resolution: 120dpi) {
-      @content;
-    }
-  }
-
-  @else if $media == oldIE {
-    @media screen and (-ms-high-contrast: active), (-ms-high-contrast: none) {
-      @content;
-    }
-  }
-}
+@mixin v-scale($property, $number) {
+  #{$property}: $number;
 
-@mixin retina {
-  @media only screen and (-webkit-min-device-pixel-ratio: 1.3),
-    only screen and (-o-min-device-pixel-ratio: 13/10),
-    only screen and (min-resolution: 120dpi) {
-    @content;
+  @supports (--css: variables) {
+    #{$property}: calc(#{$number} * var(--scale-element));
   }
 }
 
 // segment customisable per-component / instance rather than being forced to work around defaults.
-@mixin segment($padding-top: $segment-padding, $padding-bottom: $segment-padding, $padding-top-mobile: $segment-padding-mobile, $padding-bottom-mobile: $segment-padding-mobile) {
-  padding-top: $padding-top-mobile;
-  padding-bottom: $padding-bottom-mobile;
+@mixin segment (
+  $padding-top: $segment-padding,
+  $padding-bottom: $segment-padding,
+  $padding-top-mobile: $segment-padding-mobile,
+  $padding-bottom-mobile: $segment-padding-mobile) {
+
+  // set up unitless values
+  $padding-top-mobile-unitless: strip-unit($padding-top-mobile);
+  $padding-bottom-mobile-unitless: strip-unit($padding-bottom-mobile);
+  $padding-top-unitless: strip-unit($padding-top);
+  $padding-bottom-unitless: strip-unit($padding-bottom);
+
+  @include v-scale(padding-top, $padding-top-mobile-unitless * 1px);
+  @include v-scale(padding-bottom, $padding-bottom-mobile-unitless * 1px);
 
   @media (min-width: $min-tablet) {
-    padding-top: percentage(strip-unit($padding-top) / strip-unit($page-limit));
-    padding-bottom: percentage(strip-unit($padding-bottom) / strip-unit($page-limit));
+    @include v-scale(padding-top, percentage(($padding-top-unitless * 1px) / $page-limit));
+    @include v-scale(padding-bottom, percentage(($padding-top-unitless * 1px) / $page-limit));
```

Comment:
Duplicate `padding-top-unitless` here, should be `padding-bottom-unitless`

Reply:
oops. thanks again